pyomo/contrib/fbbt/interval.py

```diff
@@ -39,35 +39,59 @@ def mul(xl, xu, yl, yu):
 
 
 def inv(xl, xu, feasibility_tol):
-    if xl <= feasibility_tol and xu >= -feasibility_tol:
-        # if the denominator (x) can include 0, then 1/x is unbounded.
-        lb = -inf
+    """
+    The case where xl is very slightly positive but should be very slightly negative (or xu is very slightly negative
+    but should be very slightly positive) should not be an issue. Suppose xu is 2 and xl is 1e-15 but should be -1e-15.
+    The bounds obtained from this function will be [0.5, 1e15] or [0.5, inf), depending on the value of
+    feasibility_tol. The true bounds are (-inf, -1e15] U [0.5, inf), where U is union. The exclusion of (-inf, -1e15]
+    should be acceptable. Additionally, it very important to return [0, 1/xu] when xl is 0.
+    """
+    if xu <= 0 <= xl:
+        if abs(xl - xu) <= feasibility_tol:
+            raise IntervalException(f'Division by zero in inv; xl: {xl}; xu: {xu}')
+        else:
+            raise InfeasibleConstraintException(f'lower bound is greater than upper bound in inv; xl: {xl}; xu: {xu}')
+    elif 0 <= xl <= feasibility_tol:
+        # xu must be strictly positive
         ub = inf
-    else:
+        lb = 1.0 / xu
+    elif xl > feasibility_tol:
+        # xl and xu must be strictly positive
+        ub = 1.0 / xl
+        lb = 1.0 / xu
+    elif -feasibility_tol <= xu <= 0:
+        # xl must be strictly negative
+        lb = -inf
+        ub = 1.0 / xl
+    elif xu < -feasibility_tol:
+        # xl and xu must be strictly negative
         ub = 1.0 / xl
         lb = 1.0 / xu
+    else:
+        # everything else
+        lb = -inf
+        ub = inf
```

Comment:
Why does `[-eps, 2]` return `[-inf, inf]`, but `[eps, 2]` returns `[0.5, inf]`?

Reply:
This is just me being careful. If the lower bound is negative at all, I return `-inf` for the lower bound (1/-1e-12 is -1e12). If `xl >= -eps`, it could reasonably be assumed that the result should positive. However, it is safer to return `-inf` for the resulting lower bound.